kong/router.lua

```diff
@@ -376,7 +376,7 @@ local function marshall_route(r)
 
     local has_host_wildcard
     local has_host_plain
-    local has_port
+    local has_port, has_wildcard_host_port
```

Comment:
in this case we should let `has_port` variable live inside of for loop, line 393 becomes
```lua
local _, _, has_port = split_port(host)
```
and this line we don't declare `has_port` only `has_wildcard_host_port`.

Reply:
done.